### DISCUSSION TITLE
Ubuntu系统挂载Alist网盘 WebDav服务完全指南

### DISCUSSION THREAD

**Discussion by cat9999sss** [AlistGo/alist#5470] [General]
在数字化时代，个人搭建网盘已经非常成熟了，Alist网盘就是其中的佼佼者，已经成为数据存取的重要工具。对于Ubuntu用户来说，将Alist网盘通过WebDav协议挂载到本地，无疑可以极大提升工作效率。下面我将分步骤介绍如何在Ubuntu系统中挂载Alist WebDav。

原文地址：https://dashen.wang/5684.html ，喜欢可以支持一下。

### 安装必要的davfs2组件

在开始之前，确保你的系统已经安装了davfs2，这是一个使得Linux系统可以通过WebDav协议挂载远程网络存储的工具。打开终端，输入以下命令安装davfs2：

```bash
sudo apt update
sudo apt install davfs2 -y
```

### 配置davfs2以支持Alist WebDav

安装完成后，我们需要对davfs2进行适配Alist的配置。通过以下命令编辑davfs2的配置文件：

```bash
sudo nano /etc/davfs2/davfs2.conf
```

在配置文件中找到 `ignore_dav_header` 这一项。如果前面有注释符号，请删除它，并把该项的值设置为 `1`：

```plaintext
ignore_dav_header 1
```

保存并退出编辑器（在nano中，使用`Ctrl+O`保存，`Ctrl+X`退出）。

### 挂载Alist WebDav

接下来，使用mount命令手动挂载Alist WebDav。你需要替换以下命令中的 `ip或者域名` 和 `端口号`，以匹配你的Alist服务配置：

```bash
sudo mount -t davfs http://ip或者域名:端口号/dav /mnt/webdav
```
> 这里要注意，如果你的Alist开启了ssl，端口就是443，协议也应该改为`https://`。
`/mnt/webdav` 这个文件夹是可以自定义的。你可以放在你喜欢的任何地方。

如果你的Alist服务设置了用户名和密码，系统会提示你输入这些信息。

### 设置系统启动时自动挂载

为了在每次启动系统时自动挂载Alist WebDav，我们需要编辑 `/etc/fstab` 文件。输入以下命令来编辑该文件：

```bash
sudo nano /etc/fstab
```

在文件的最后添加以下内容，注意替换 `ip或者域名` 和 `端口号`：

```plaintext
http://ip或者域名:端口号/dav /mnt/webdav davfs _netdev,auto 0 0
```

这样设置后，每次Ubuntu启动时，都会自动尝试挂载Alist WebDav到 `/mnt/webdav`。

以上就是将Alist网盘通过WebDav协议挂载到Ubuntu系统的详细步骤。只需简单的几步，你就可以在本地系统中无缝访问Alist网盘中的文件，提高日常工作的便利性和效率。

**Comment by hankqqu** [comment]
> 在数字化时代，个人搭建网盘已经非常成熟了，Alist网盘就是其中的佼佼者，已经成为数据存取的重要工具。对于Ubuntu用户来说，将Alist网盘通过WebDav协议挂载到本地，无疑可以极大提升工作效率。下面我将分步骤介绍如何在Ubuntu系统中挂载Alist WebDav。
> 
> 
> 
> 原文地址：https://dashen.wang/5684.html ，喜欢可以支持一下。
> 
> 
> 
> ### 安装必要的davfs2组件
> 
> 
> 
> 在开始之前，确保你的系统已经安装了davfs2，这是一个使得Linux系统可以通过WebDav协议挂载远程网络存储的工具。打开终端，输入以下命令安装davfs2：
> 
> 
> 
> ```bash
> 
> sudo apt update
> 
> sudo apt install davfs2 -y
> 
> ```
> 
> 
> 
> ### 配置davfs2以支持Alist WebDav
> 
> 
> 
> 安装完成后，我们需要对davfs2进行适配Alist的配置。通过以下命令编辑davfs2的配置文件：
> 
> 
> 
> ```bash
> 
> sudo nano /etc/davfs2/davfs2.conf
> 
> ```
> 
> 
> 
> 在配置文件中找到 `ignore_dav_header` 这一项。如果前面有注释符号，请删除它，并把该项的值设置为 `1`：
> 
> 
> 
> ```plaintext
> 
> ignore_dav_header 1
> 
> ```
> 
> 
> 
> 保存并退出编辑器（在nano中，使用`Ctrl+O`保存，`Ctrl+X`退出）。
> 
> 
> 
> ### 挂载Alist WebDav
> 
> 
> 
> 接下来，使用mount命令手动挂载Alist WebDav。你需要替换以下命令中的 `ip或者域名` 和 `端口号`，以匹配你的Alist服务配置：
> 
> 
> 
> ```bash
> 
> sudo mount -t davfs http://ip或者域名:端口号/dav /mnt/webdav
> 
> ```
> 
> > 这里要注意，如果你的Alist开启了ssl，端口就是443，协议也应该改为`https://`。
> 
> `/mnt/webdav` 这个文件夹是可以自定义的。你可以放在你喜欢的任何地方。
> 
> 
> 
> 如果你的Alist服务设置了用户名和密码，系统会提示你输入这些信息。
> 
> 
> 
> ### 设置系统启动时自动挂载
> 
> 
> 
> 为了在每次启动系统时自动挂载Alist WebDav，我们需要编辑 `/etc/fstab` 文件。输入以下命令来编辑该文件：
> 
> 
> 
> ```bash
> 
> sudo nano /etc/fstab
> 
> ```
> 
> 
> 
> 在文件的最后添加以下内容，注意替换 `ip或者域名` 和 `端口号`：
> 
> 
> 
> ```plaintext
> 
> http://ip或者域名:端口号/dav /mnt/webdav davfs _netdev,auto 0 0
> 
> ```
> 
> 
> 
> 这样设置后，每次Ubuntu启动时，都会自动尝试挂载Alist WebDav到 `/mnt/webdav`。
> 
> 
> 
> 以上就是将Alist网盘通过WebDav协议挂载到Ubuntu系统的详细步骤。只需简单的几步，你就可以在本地系统中无缝访问Alist网盘中的文件，提高日常工作的便利性和效率。

**Comment by GuoMieClub** [comment]
成功挂载并可读取出文件列表，但只可写入，无法读取/写出，提示input/output error。文件权限已为777，use_locks为0

**Reply by DVKunion** [reply]
the same +1, any ideas?

**Reply by GuoMieClub** [reply]
换rclone 解决

**Reply by wt666666** [reply]
> the same +1, any ideas?

The same +1, 解决了吗?

**Comment by TimeController3** [comment]
rclone不就行了吗

**Comment by ixiaan** [comment]
如果是LXC，提示can't open fuse device
则运行
`mknod -m 666 /dev/fuse c 10 229`
因fuse模块未被LXC调用

**Reply by ixiaan** [reply]
1、进入母鸡 1.1、以root身份运行
要在母鸡下中获取你的小鸡编号例如：102.conf，请按以下方式编辑102.conf
这里102.conf是你小鸡的编号

nano /etc/pve/nodes/你的host名称/lxc/102.conf
nano /etc/pve/lxc/102.conf

#最下面加入
#mp0: /root/gg-web-bak,mp=/webbak
lxc.mount.entry = /dev/fuse dev/fuse none bind,create=file
1.2、重启小鸡
mknod -m 666 /dev/fuse c 10 229

**Reply by ixiaan** [reply]
或者把这个脚本写入开机自启动
`nano /etc/init.d/addfuse.sh `

写入
`#! /bin/sh`


`mknod -m 666 /dev/fuse c 10 229`
保存

赋权
`chmod +x /etc/init.d/addfuse.sh`

加入启动（软连接到rc2.d）
`ln -s /etc/init.d/addfuse.sh /etc/rc2.d/S01addfuse`
`reboot`
重启测试

**Comment by ixiaan** [comment]
添加一个保存密码的办法
`vi /etc/davfs2/davfs2.conf`
找到“use_locks”，去掉前面的 # 然后把 1 改为 0 后保存。

把密码保存到文件
`echo "http://ip或者域名:端口号/dav user password" > /etc/davfs2/secrets`

之后再如上文叙述的设定开机自动挂载即可

**Comment by zxc100201** [comment]
要么在alist中编辑挂载信息的WebDAV策略为本地代理，要么修改davfs2.conf配置文件，添加 follow_redirect 1 参数

**Reply by zxc100201** [reply]
针对成功挂载并可读取出文件列表，但只可写入，无法读取/写出，提示input/output error的情况

**Reply by IhaveAyellowDog** [reply]
> 要么在alist中编辑挂载信息的WebDAV策略为本地代理，要么修改davfs2.conf配置文件，添加 follow_redirect 1 参数

牛逼，差点放弃了，全网搜索下来，这个是找对问题了的！！

**Reply by TTITion** [reply]
可以了！谢谢~~

**Reply by lhuanyun** [reply]
学习一下

**Reply by tigeryzx** [reply]
非常感谢，全网找遍，只有这个才能这个问题
